classes/class-wc-connect-notice-manager.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+// No direct access please
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+if ( ! class_exists( 'WC_Connect_Notice_Manager' ) ) {
+
+	class WC_Connect_Notice_Manager {
```

Comment:
Takes care of getting a list of notices and allowing consuming code to dismiss notices. A notices repository of sorts.